flink-ml-iteration/src/main/java/org/apache/flink/ml/iteration/operator/coordinator/HeadOperatorCoordinator.java

```diff
@@ -63,10 +63,16 @@ public void subtaskReady(int i, SubtaskGateway subtaskGateway) {
     }
 
     @Override
-    public void resetToCheckpoint(long l, @Nullable byte[] bytes) {}
+    public void resetToCheckpoint(long l, @Nullable byte[] bytes) {
+        for (int i = 0; i < context.currentParallelism(); ++i) {
```

Comment:
Guess we should either remove the `for` loop or add `i` to the parameter list of `removeProgressInfo`.